quote-server/server.js

```diff
@@ -1,32 +1,33 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
 import quotes from "./quotes.json" assert { type: "json" };
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
+app.use(cors());
+
 app.get("/", (request, response) => {
   response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
 });
 
-//START OF YOUR CODE...
+app.get("/quotes", (req, res) => {
+  res.json(quotes);
+});
+
+app.get("/quotes/random", (req, res) => {
+  res.json(pickFromArray(quotes));
+});
 
-//...END OF YOUR CODE
+app.get("/quotes/search", (req, res) => {
+  const searchTerm = req.query.term;
+  if (!searchTerm) {
+    return res.status(400).json({ error: "No search term provided" });
+  }
+  const searchedQuotes = quotes.filter((quote) => quote.quote.toLowerCase().includes(searchTerm.toLowerCase() || quote.author.toLowerCase().includes(searchTerm.toLowerCase())));
```

Comment:
How many times will this end up calling `searchTerm.toLowerCase()`? Does the value returned by that function change between calls? Can you think of a way to call this function less often?